code/game/objects/items/weapons/shields.dm

```diff
@@ -35,27 +35,102 @@
+	trait_slots = list(
+		slot_l_hand,
+		slot_r_hand
+	)
+
+	equipped_traits = list(TRAIT_UNPUSHABLE)
+
+/obj/item/shield/Initialize(mapload, ...)
+	. = ..()
+	shield_health = max_shield_health
+
+/obj/item/shield/get_examine_text(mob/user, distance, is_adjacent, infix, suffix, get_extended)
+	. = ..()
+	. += get_durability_examine_text(user, distance, is_adjacent, infix, suffix, get_extended)
+
+/obj/item/shield/proc/get_durability_examine_text(mob/user, distance, is_adjacent, infix, suffix, get_extended)
+	. = list()
+	if(is_adjacent)
+		. += SPAN_NOTICE("\The [src] has [SPAN_BOLD("[shield_health]")]/[SPAN_BOLD("[max_shield_health]")] durability remaining.")
+		. += SPAN_NOTICE("\The [src] can be repaired with [SPAN_BOLD("[material_repair_type]")].") // TO-DO
+
+/obj/item/shield/attackby(obj/item/attacking_item, mob/user)
+	if(material_repair_type && istype(attacking_item, /obj/item/stack/material))
+		var/obj/item/stack/material/material_stack = attacking_item
+		if(material_stack.material.name != material_repair_type)
+			return
+		if(shield_health == max_shield_health)
+			to_chat(user, SPAN_WARNING("\The [src] is already fully repaired!"))
+			return
+		var/amount_to_use = 3
+		if(!material_stack.can_use(amount_to_use))
+			to_chat(user, SPAN_WARNING("You don't have enough material to repair \the [src]!"))
+			return
+		user.visible_message("[SPAN_BOLD("[user]")] starts repairing \the [src] with \the [material_stack]...", SPAN_NOTICE("You start repairing \the [src] with \the [material_stack]."))
+		if(do_after(user, 20 SECONDS, src, do_flags = DO_UNIQUE))
+			if(!material_stack.can_use(amount_to_use))
+				to_chat(user, SPAN_WARNING("You don't have enough material to repair \the [src]!"))
+				return
+			material_stack.use(amount_to_use)
+			shield_health = clamp(shield_health + (max_shield_health / 3), 0, max_shield_health)
+			user.visible_message("[SPAN_BOLD("[user]")] [shield_health != max_shield_health ? "partially repairs" : "repairs"] \the [src] with \the [material_stack]...", SPAN_NOTICE("You [shield_health != max_shield_health ? "partially repair" : "repair"] \the [src] with \the [material_stack]."))
+			update_icon()
+		return
+	return ..()
+
+/obj/item/shield/handle_shield(mob/user, var/on_back, var/damage, obj/damage_source = null, mob/attacker = null, def_zone = null, attack_text = "the attack")
 	var/shield_dir = on_back ? user.dir : GLOB.reverse_dir[user.dir]
 
 	if(user.incapacitated() || !(check_shield_arc(user, shield_dir, damage_source, attacker)))
 		return FALSE
 
-	if(prob(get_block_chance(user, damage, damage_source, attacker)))
+	var/datum/component/armor/armor_datum = GetComponent(/datum/component/armor)
+	var/list/damage_args = armor_datum.apply_damage_modifications(damage, damage_source?.damage_type() || DAMAGE_BRUTE, damage_source?.damage_flags() || 0, user, damage_source?.armor_penetration || 0, TRUE)
+	handle_damage(user, damage_args[1])
```

Comment:
Shields should probably do the same thing barricades do where projectiles have a % chance to pass through them depending on current health (from 0% up to maybe 50 to 75%) which is more fair than a consistent 100%. 

I'm also not convinced that a shield with no bullet armour should eat bullets. If a shield doesn't have enough armour to lower the damage below a certain value, it shouldn't eat the damage at all, or at the very least it should be MUCH stronger for the purposes of breaking the shield. My proposition is making all of these depend on damage type.

- Blunt non-sharp non-edge weapons should be really good at breaking shields.
- Edge damage should probably not do that much - not a good idea to bring a knife to a shield fight.
- Burn weapons should be about as effective if not more than blunt weapons.
- Projectiles and piercing weapons should be able to go THROUGH shields with the aforementioned % chance depending on current shield durability.